arelle/plugin/validate/ESEF/ESEF_Current/DTS.py

```diff
@@ -108,7 +109,7 @@ def lc3wordAdjust(word: str) -> str:
                         val.modelXbrl.error("ESEF.3.4.3.extensionTaxonomyDimensionNotAssignedDefaultMemberInDedicatedPlaceholder",
                             _("Each dimension in an issuer specific extension taxonomy MUST be assigned to a default member in the ELR with role URI http://www.esma.europa.eu/xbrl/role/core/ifrs-dim_role-990000 defined in esef_cor.xsd schema file. %(qname)s"),
                             modelObject=modelConcept, qname=modelConcept.qname)
-                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and modelConcept.typeQname not in qnDomainItemTypes:
+                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and ((is2022DisclosureSystem(val.modelXbrl) and modelConcept.typeQname not in qnDomainItemTypes) or (not is2022DisclosureSystem(val.modelXbrl) and modelConcept.typeQname not in qnDomainItemTypes2023)):
```

Comment:
Suggest moving the year disclosure system selection out of the if statement:
```suggestion
                    esefDomainItemTypes = qnDomainItemTypes if getDisclosureSystemYear(val.modelXbrl) < 2023 else qnDomainItemTypes2023
                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and modelConcept.typeQname not in esefDomainItemTypes:
```